media/js/firefox/firstrun/membership.js

```diff
@@ -0,0 +1,214 @@
+        var iframe = document.getElementById('fxa');
+        var dataSrc = iframe.getAttribute('data-src');
+        iframe.setAttribute('data-src', dataSrc.replace('utm_campaign=fxa-embedded-form', 'utm_campaign=' + utmCampaign));
+    }
+
+    // animates page to step2 and loads the FxA sign up form.
+    function goToStep2() {
+
+        // fancy transision requires Web Animations and Promise support.
+        if (cutsTheMustard) {
+            fadeOutStep1().then(function() {
+                // destroy slider on step-1 as no longer needed.
+                $slideShow.cycle('destroy');
+                // fade in step 2.
+                fadeInStep2().then(initAccountsForm).catch(function(reason) {
+                    throw new Error(reason);
+                });
+            }).catch(function(reason) {
+                throw new Error(reason);
+            });
+        }
+        // fallback to simple non-animated transition.
+        else {
+            step1.classList.add('hidden');
+            $slideShow.cycle('destroy');
+
+            step2.classList.remove('hidden');
+            initAccountsForm();
+        }
+    }
+
+    function onButtonClick(e) {
+        e.preventDefault();
+        updateUTMParam(e.target.dataset.id);
+        unbindEvents();
+        goToStep2();
+
+        window.dataLayer.push({
+            'event': 'in-page-interaction',
+            'eAction': 'Button Click',
+            'eLabel': e.target.dataset.label
+        });
+    }
+
+    function onButtonMouseOver(e) {
+        $slideShow.cycle('pause'); //pause auto-running slideshow
+        $slideShow.cycle('goto', e.target.dataset.index); //go to specific path
+    }
+
+    function onButtonMouseOut() {
+        $slideShow.cycle('resume'); //resume auto-running slideshow
+    }
+
+    function onLaterLinkClick(e) {
+        e.preventDefault();
+        window.dataLayer.push({
+            'event': 'in-page-interaction',
+            'eAction': 'Link Click',
+            'eLabel': 'Remind Me Later'
+        });
+    }
+
+    function bindEvents() {
+        for (var i = 0; i < button.length; i++) {
+            button[i].addEventListener('mouseover', onButtonMouseOver, false);
+            button[i].addEventListener('mouseout', onButtonMouseOut, false);
+            button[i].addEventListener('click', onButtonClick, false);
+        }
+
+        laterLink.addEventListener('click', onLaterLinkClick, false);
+    }
+
+    function unbindEvents() {
+        for (var i = 0; i < button.length; i++) {
+            button[i].removeEventListener('mouseover', onButtonMouseOver, false);
+            button[i].removeEventListener('mouseout', onButtonMouseOut, false);
+            button[i].removeEventListener('click', onButtonClick, false);
+        }
+
+        laterLink.removeEventListener('click', onLaterLinkClick, false);
+    }
+
+    function animateElement(selector, keyframes, options) {
```

Comment:
This is my first forray into the Web Animations API. This is just for a single experiment right now, but I may refactor some of what's here into a nice little JS animation library in the future (as some form of personal deliverable)